File: backend/src/main/java/com/sicpr/backend/mensagem/model/Mensagem.java
package com.sicpr.backend.mensagem.model;

import com.sicpr.backend.user.model.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "mensagens")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Mensagem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "remetente_id", nullable = false)
    private User remetente;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "destinatario_id", nullable = false)
    private User destinatario;

    @Column(columnDefinition = "TEXT")
    private String texto;

    private String anexoNomeOriginal;

    private String anexoNomeArquivo;

    private String anexoContentType;

    private Long anexoTamanho;

    private Boolean lida;

    private LocalDateTime criadoEm;

    private LocalDateTime expiraEm;

    @PrePersist
    protected void onCreate() {
        criadoEm = LocalDateTime.now();
        expiraEm = criadoEm.plusHours(24);
        if (lida == null) lida = false;
    }
}

File: frontend/app/(pages)/mensagens/types/mensagem.ts
// app/mensagens/types/mensagem.ts
export interface MensagemUser {
  id: number;
  username: string;
  nomeCompleto: string | null;
  perfil: string | null;
  status: string;
}

export interface Mensagem {
  id: number;
  remetenteId: number;
  remetenteNome: string;
  remetenteCargo: string | null;
  destinatarioId: number;
  destinatarioNome: string;
  destinatarioCargo: string | null;
  texto: string | null;
  anexoNomeOriginal: string | null;
  anexoContentType: string | null;
  anexoTamanho: number | null;
  anexoUrl: string | null;
  lida: boolean;
  criadoEm: string;
  expiraEm: string;
}
